management/flow_test.go

```diff
@@ -0,0 +1,269 @@
+package management
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/auth0/go-auth0"
+)
+
+/*
+Flow tests.
+*/
+func TestFlowManager_Create(t *testing.T) {
+	configureHTTPTestRecordings(t)
+	flow := &Flow{
+		Name:        auth0.String("test-flow"),
+		Description: auth0.String("A test flow"),
+		Synchronous: auth0.Bool(true),
+	}
+
+	err := api.Flow.Create(context.Background(), flow)
+
+	assert.NoError(t, err)
+	assert.NotEmpty(t, flow.GetID())
+
+	t.Cleanup(func() {
+		cleanupFlow(t, flow.GetID())
+	})
+}
+
+func TestFlowManager_Read(t *testing.T) {
+	configureHTTPTestRecordings(t)
+	expectedFlow := givenAFlow(t)
+
+	actualFlow, err := api.Flow.Read(context.Background(), expectedFlow.GetID())
+
+	assert.NoError(t, err)
+	assert.Equal(t, expectedFlow, actualFlow)
+}
+
+func TestFlowManager_Update(t *testing.T) {
+	configureHTTPTestRecordings(t)
+	expectedFlow := givenAFlow(t)
+	updatedFlow := &Flow{
+		Description: auth0.String("Updated flow description test"),
+	}
+	err := api.Flow.Update(context.Background(), expectedFlow.GetID(), updatedFlow)
+
+	assert.NoError(t, err)
+	assert.Equal(t, "Updated flow description test", updatedFlow.GetDescription())
+	assert.Equal(t, expectedFlow.GetName(), updatedFlow.GetName())
+}
+
+func TestFlowManager_Delete(t *testing.T) {
+	configureHTTPTestRecordings(t)
+	expectedFlow := givenAFlow(t)
+
+	err := api.Flow.Delete(context.Background(), expectedFlow.GetID())
+	assert.NoError(t, err)
+}
+
+func TestFlowManager_List(t *testing.T) {
+	configureHTTPTestRecordings(t)
+	flow := givenAFlow(t)
+
+	flowList, err := api.Flow.List(context.Background())
+	assert.NoError(t, err)
+	assert.Greater(t, len(flowList.Flows), 0)
+	assert.Contains(t, getFlowIDs(flowList.Flows), flow.GetID())
```

Comment:
We can do it this way, so we won't need to add a new function, and this will also check other attributes in the body as well.
```
func TestFlowManager_List(t *testing.T) {
	configureHTTPTestRecordings(t)
	flow := givenAFlow(t)
	flow.Actions = nil
	flow.Triggers = nil
	flow.Security = nil
	flowList, err := api.Flow.List(context.Background())
	assert.NoError(t, err)
	assert.Greater(t, len(flowList.Flows), 0)
	assert.Contains(t, flowList.Flows, flow)
}
```